webapp/src/js/router.js

```diff
@@ -146,12 +146,18 @@
         }
       })
       .state('contacts.addChild', {
-        url: '/:parent_id/add/:type',
+        url: '/:parent_id/add/:type/:fromList',
```

Comment:
I feel like this should be a query parameter, eg: `url: '/:parent_id/add/:type?from',`.

Reply:
Done, I've switched it to a `from` query param and check for `list` value in contacts-edit controller.